### PR TITLE
Influxdb Support

### PR DESCRIPTION
**Pull request by Sharpie**
This changeset incorporates the work from PR #18 along with some additional enhancement and fixups. Also intend to re-name the `json2graphite.rb` script to `json2metrics.rb`, possibly with `json2graphite.rb` left behind as a symlink. Left that out of this changeset as it causes GitHub to "simplify" the dif into a delete + new file which covers up the actual changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,28 +21,46 @@ You can then view the metrics by visting `http://localhost:3000` in your browser
...
+InfluxDB output can be produced using the `--convert-to` flag:
+
+```
+./json2graphite.rb ~/Downloads/logdump/puppetserver/*.json --convert-to influxdb
+```
+
+When `--netcat` is used with InfluxDB output, data will be sent to port 8086. The `--influx-db` flag must also be used to specify a database to write to:
+
+```
+./json2graphite.rb ~/Downloads/logdump/puppetserver/*.json --convert-to influxdb --netcat localhost --inflix-db pe-metrics
```
Comment (jarretlavallee) [inline README.md]: Small typo in `--inflix-db`

**Review by jarretlavallee — APPROVED** [review]
👍  Great work on this. I tested against the pe_metrics_dashboard module and it is working with the existing graphs. The output looks good and is easy to use. Thanks for putting in the work on this, especially around the tags!

**Comment by suckatrash** [comment]
Looks great to me!  Looks like we're getting all the resource / http / function metrics so we can begin building out puppetserver workload dashboards in the pe_metrics_dashboard project 💯

File: README.md (hunk 1 above)
Comment (Sharpie) [reply README.md]: Ah, good catch. I'll get that cleaned up.

**Comment by Sharpie** [comment]
@suckatrash One change I've been pondering that would affect the dashboard templates is to remove the current duplication that exists between the measurement name and field name.

E.g., currently we take a metric like PuppetDB fact storage time and split each different value into completely separate measurements:

```
storage_replace-facts-time.Max,server=127-0-0-1,service=puppetdb Max=3111.870449 1510476001
storage_replace-facts-time.50thPercentile,server=127-0-0-1,service=puppetdb 50thPercentile=1187.890588 1510476001
storage_replace-facts-time.95thPercentile,server=127-0-0-1,service=puppetdb 95thPercentile=1565.707702 1510476001
...
```

Which means that in order to flip a graph between the Max or 95thPercentile, one has to change the measurement selection from `storage_replace-facts-time.Max` to `storage_replace-facts-time.95thPercentile` and then change the field selection again from `Max` to `95thPercentile`.

We could cut half of that work by not duplicating the field name as part of the measurement name:

```
storage_replace-facts-time,server=127-0-0-1,service=puppetdb Max=3111.870449 1510476001
storage_replace-facts-time,server=127-0-0-1,service=puppetdb 50thPercentile=1187.890588 1510476001
storage_replace-facts-time,server=127-0-0-1,service=puppetdb 95thPercentile=1565.707702 1510476001
...
```

**Comment by suckatrash** [comment]
@Sharpie I'm ok with changing the measurements.  Sounds like it will simplify things in the long run.

**Review by suckatrash — APPROVED** [review]

**Comment by npwalker** [comment]
@Sharpie is this good to be merged or are there further changes needed?

**Comment by Sharpie** [comment]
I think it's good to merge as-is. I'll open a separate PR for flattening the metric names.

File: README.md (hunk 1 above)
Comment (npwalker) [reply README.md]: @Sharpie first the typo up and I'll merge

Comment (Sharpie) [reply README.md]: Done!

**Pull request merged**